pkg/cluster/install.go

```diff
@@ -156,6 +161,17 @@ func (m *manager) adminUpdate() []steps.Step {
 	return toRun
 }
 
+func (m *manager) shouldUpdateOperator() bool {
+	runningVersion, err := semver.NewVersion(m.doc.OpenShiftCluster.Properties.ClusterProfile.Version)
```

Comment:
I know it's convention to use the cluster doc, but Is it possible to look directly at the cluster version operator resources to get the running version rather than use the doc?

My concern is just around potential drift or a race between the version in-cluster and the version reporting in the cluster doc. If the cluster version in the doc is guaranteed to be current by the time this step runs, I think we're ok to leave it as-is. It's not totally clear to me looking at code where we update the doc with the current running version.

Reply:
I had that thought too. I can investigate this further.

Reply:
So the version check here is done at the time the steps are defined, not when they are executed, so it's basically guaranteed that it will not be up to date since the check is done before any actions are performed. 

Because of this, I'm also hesitant to add any more complicated checks to e.g. the CVO resource at this stage. I think a better approach would be to do these checks within each action at run-time, and/or potentially grouping the operator reconciliation steps into a single Action with its own sub-task runner, that does the checks before execution? 

Either way would be a larger change, so I'm curious to explore the impact of the current implementation to see if we can make this improvement a post-merge activity. As far as I see it, if we PUCM a cluster that was on <=4.6 during our last PUCM but has since been upgraded, they would lose the operator upgrade from that PUCM and would have to wait for the next PUCM to be performed in order to upgrade.

Reply:
Never mind, the version update is done by the frontend, before the backend even starts defining which steps to run: 
- (frontend enrichment) https://github.com/Azure/ARO-RP/blob/master/pkg/frontend/openshiftcluster_putorpatch.go#L120 
- (actual version update) https://github.com/Azure/ARO-RP/blob/master/pkg/util/clusterdata/cluster_version.go#L22